test/integration/targets/docker_network/tasks/tests/ipam.yml

```diff
@@ -0,0 +1,212 @@
+
+- name: Create network with deprecated custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_0 }}"
+    ipam_options:
+      subnet: 172.3.29.0/24
+  register: network
+
+- assert:
+    that:
+      - network is changed
+
+- name: Change subnet of network with deprecated custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_0 }}"
+    ipam_options:
+      subnet: 172.3.30.0/24
+  register: network
+  diff: yes
+
+- assert:
+    that:
+      - network is changed
+      - network['diff'] | length == 1
+      - network['diff'][0] == "ipam_config[0].subnet"
+
+- name: Cleanup network with ipam_config and deprecated ipam_options
+  docker_network:
+    name: "{{ nname_ipam_0 }}"
+    state: absent
+
+
+#################### network-ipam-1 ####################
+- name: Create network with custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_1 }}"
+    ipam_config:
+      - subnet: 172.3.27.0/24
+        gateway: 172.3.27.2
+        iprange: 172.3.27.0/26
+        aux_addresses:
+          host1: 172.3.27.3
+          host2: 172.3.27.4
+  register: network
+
+- assert:
+    that:
+      - network is changed
+
+- name: Change subnet, gateway, iprange and auxiliary addresses of network with custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_1 }}"
+    ipam_config:
+      - subnet: 172.3.28.0/24
+        gateway: 172.3.28.2
+        iprange: 172.3.28.0/26
+        aux_addresses:
+          host1: 172.3.28.3
+  register: network
+  diff: yes
+
+- assert:
+    that:
+      - network is changed
+      - network['diff'] | length == 4
+      - '"ipam_config[0].subnet" in network["diff"]'
+      - '"ipam_config[0].gateway" in network["diff"]'
+      - '"ipam_config[0].iprange" in network["diff"]'
+      - '"ipam_config[0].aux_addresses" in network["diff"]'
+
+- name: Remove gateway and iprange of network with custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_1 }}"
+    ipam_config:
+      - subnet: 172.3.28.0/24
+  register: network
+
+- assert:
+    that:
+      - network is not changed
+
+- name: Cleanup network with custom IPAM config
+  docker_network:
+    name: "{{ nname_ipam_1 }}"
+    state: absent
+
+
+#################### network-ipam-2 ####################
+
+- name: Create network with IPv6 IPAM config
+  docker_network:
+    name: "{{ nname_ipam_2 }}"
+    enable_ipv6: yes
+    ipam_config:
+      - subnet: fdd1:ac8c:0557:7ce0::/64
+  register: network
+
+- assert:
+    that:
+      - network is changed
+
+- name: Change subnet of network with IPv6 IPAM config
+  docker_network:
+    name: "{{ nname_ipam_2 }}"
+    enable_ipv6: yes
+    ipam_config:
+      - subnet: fdd1:ac8c:0557:7ce1::/64
+  register: network
+  diff: yes
+
+- assert:
+    that:
+      - network is changed
+      - network['diff'] | length == 1
+      - network['diff'][0] == "ipam_config[0].subnet"
+
+- name: Change subnet of network with IPv6 IPAM config
```

Comment:
[minor] This test's name should call out that it's testing the removal of the netmask. It'll make it easier to figure out what's failing if this test ever finds a regression.

Reply:
It's not testing removal of the netmask, but produces a failure. The netmask must always be there for it to work. But yes, a more expressive name would be nice. But that can wait until another PR, let's get this one merged first :)

Reply:
It's testing that removing the netmask produces the expected failure, hence why it should have a more expressive name. But as you say, it should be merged, which is why I marked my shipit in another comment on the main thread.

Reply:
Ah, sorry, I think I misunderstood you :) Anyway, I think we fully agree ;)